client/fingerprint_manager.go

```diff
@@ -361,9 +359,30 @@ func (fm *FingerprintManager) fingerprintDriver(name string, f fingerprint.Finge
 	delete(di.Attributes, driverKey)
 
 	if node := fm.updateNodeFromDriver(name, di, nil); node != nil {
-		fm.nodeLock.Lock()
-		fm.node = node
-		fm.nodeLock.Unlock()
+		fm.setNode(node)
+	}
+
+	// Get a copy of the current node
+	var driverExists, driverIsHealthy bool
+	fm.nodeLock.Lock()
+	driverInfo, driverExists := fm.node.Drivers[name]
+	if driverExists {
+		driverIsHealthy = driverInfo.Healthy
+	}
+	fm.nodeLock.Unlock()
+
+	// If either 1) the driver is undetected or 2) if the driver does not have
+	// periodic health checks enabled, set the health status to the match that
+	// of the fingerprinter
+	if !hasPeriodicHealthCheck || !response.Detected && driverExists && driverIsHealthy {
+		healthInfo := &structs.DriverInfo{
+			Healthy:           response.Detected,
+			HealthDescription: fmt.Sprintf("Driver %s is detected: %t", name, response.Detected),
```

Comment:
Why not a health status of `fmt.Sprintf("Driver %s is not detected", name)`

Reply:
The health status will get set to match the detected status if the health check is not enabled to run periodically.

Reply:
That case doesn't really make sense? There is no non-periodic health checker: https://github.com/hashicorp/nomad/blob/a40750e596cdf6fb4b591d1c357d760d6c0bbb43/client/fingerprint_manager.go#L292-L296